app/pods/newsletter/nota-updates/route.js

```diff
@@ -28,84 +25,49 @@ export default class NewsletterNotaUpdatesRoute extends Route {
   }
 
   async model(params) {
-    let notas = A([]);
+    const processedNotas = A([]);
     const newsletterModel = this.modelFor('newsletter');
     const meeting = newsletterModel.meeting;
     const agenda = newsletterModel.agenda;
     const agendaId = agenda.id;
     const meetingId = meeting.id;
-    const agendaitemsForAgenda = await agenda.get('agendaitems');
-    // Omzetten van proxyarray naar default JS array
-    const agendaitemsArray = agendaitemsForAgenda.toArray();
-    for (const agendaitem of agendaitemsArray) {
-      if (agendaitem.showAsRemark) {
-        continue;
-      }
-      const agendaitemPriority = agendaitem.get('priority');
-      const agendaitemId = agendaitem.get('id');
-      const agendaitemShortTitle = agendaitem.get('shortTitle');
-
-      // Documenten
-      const notasOfAgendaitem = await this.store.query('piece', {
-        'filter[agendaitem][:id:]': agendaitemId,
-        'filter[document-container][type][:id:]': CONFIG.notaID,
-      });
-      if (notasOfAgendaitem.length) {
-        const lastNotaVersion = notasOfAgendaitem.firstObject;
-        const documentContainer = await lastNotaVersion.get('documentContainer');
-        const allNotaPieces = await documentContainer.get('pieces');
-        const piecesOfAgendaitemArray = allNotaPieces.toArray();
-
-        if (piecesOfAgendaitemArray) {
-          for (const piece of piecesOfAgendaitemArray) {
-            const pieceData = await NewsletterNotaUpdatesRoute.getPieceData(piece);
-            if (pieceData) {
-              const nota =  {
-                meetingId,
-                agendaId,
-                agendaitemId,
-                agendaitemPriority,
-                agendaitemShortTitle,
-                ...pieceData,
-              };
-              notas.pushObject(nota);
-            }
-          }
-        }
-      }
-    }
-    if (params.sort.includes('modified')) {
-      notas = notas.sortBy('modified');
-      if (params.sort.startsWith('-')) {
-        notas.reverseObjects();
-      }
-    }
-    const pagination = getPaginationMetadata(0, notas.length, notas.length);
-    notas.meta = {
-      count: notas.length,
-      pagination: pagination,
-    };
-    return hash({
-      notas: notas,
+    const notas = await this.store.query('piece', {
+      'filter[agendaitem][agenda][:id:]': agendaId,
+      'filter[agendaitem][show-as-remark]': false,
+      'filter[document-container][type][:id:]': CONFIG.notaID,
+      include: 'agendaitem',
+      'fields[agendaitem]': 'id,priority,short-title',
+      'fields[piece]': 'id,name,modified',
+      'page[size]': 50,
```

Comment:
Lopen we hiermee niet het risico dat we snel tegen deze limiet botsen en ze niet alle content zien die ze willen zien?
Ik zou hier voor een groter aantal gaan.

Reply:
Ik heb het getest met meer dan 50 documentversies.. ik zie het niet gebeuren dat er meer dan 15 documentversies kunnen zijn dus of dat de page size op 50 of op 100+ staat zal niet snel voorkomen.. in dat opzicht ziet 50 wel ok uit.

We kunnen 2 kanten uitgaan, ofwel laten staan of verhogen.. maar dat lijkt me eerder een business vraag waarop zij moeten antwoorden.. ik zeg door laten gaan met 50, als we ertegen lopen of we krijgen een direct number van de business kant, dan passen we het aan.